lib/urbanopt/geojson/geo_file.rb

```diff
@@ -69,15 +67,58 @@ def self.from_file(path)
           raise "GeoJSON file '#{path}' does not exist"
         end
 
-        geojson = JSON.parse(
+        geojson_file = JSON.parse(
           File.open(path, 'r', &:read),
           symbolize_names: true
         )
-        return new(geojson, path)
+
+        # validate geojson file against schema
+        geojson_errors = validate(@@geojson_schema, geojson_file)
+        unless geojson_errors.empty?
+          raise "GeoJSON file does not adhere to the schema: \n #{geojson_errors.join('\n  ')}"
+        end
+
+        # validate each feature against schema
+        geojson_file[:features].each do |feature|
+          properties = feature[:properties]
+          type = properties[:type]
+
+          errors = []
+
+          case type
+          when 'Building'
+            # Incase detailed_model_filename present check for fewer properties
+            if feature[:properties][:detailed_model_filename]
```

Comment:
@kflemin This checks whether the detailed_model_filename is present and only checks the id, name properties if it is. Othervise validates the geojson file against the buiding schema.

Reply:
looks good!